sw/host/opentitanlib/src/app/ioexpander.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::app::config;
+use crate::app::{PinStrapping, TransportWrapper};
+use crate::io::gpio::{GpioError, GpioPin, PinMode, PullMode};
+use crate::io::i2c::{self, Bus};
+use crate::transport::TransportError;
+
+use anyhow::{bail, Result};
+use std::rc::Rc;
+use std::vec::Vec;
+
+pub struct IoExpander {
+    pub pins: Vec<Rc<dyn GpioPin>>,
+}
```

Comment:
I feel like this belongs in `src/io/...` somewhere.

Reply:
I have created an `src/io/ioexpander.rs` to contain this struct.  I am not sure I think this is the best place, though, since this is not really for transports to implement.  It could be merged into the new `src/transport/ioexpander/mod.rs` file.